### PR TITLE
Added word validation input accept only alphabets , create validation…

### PR DESCRIPTION
**Pull request by mahaelbasuony**
….service and error-message.component

### CODE REVIEW

File: src/app/components/add-word/add-word.component.html [hunk 1]
```diff
@@ -13,23 +13,26 @@ <h1>
     </div>
   </h1>
   
-  <form name="formAddWord" action="" #form= "ngForm" novalidate>
+  <form name="formAddWord" action="" #form= "ngForm" novalidate [formGroup]="userForm">
     <div class="form-group">
-      <input type="text" name="word" class="form-control" [(ngModel)]="word.word" required />
+      <input type="text" formControlName="word"  name="word" class="form-control" id="word" [(ngModel)]="word.word"  required />
```
Comment (medhatdawoud) [inline src/app/components/add-word/add-word.component.html]: we are not using ID property any where, remove it and use some other selector

File: src/app/services/validation.service.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+export class ValidationService {
+  static getValidatorErrorMessage(validatorName: string, validatorValue?: any) {
+    let config = {
```
Comment (medhatdawoud) [inline src/app/services/validation.service.ts]: change the name to a meaningful name like "error messages" not config

File: src/app/services/validation.service.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      'required': 'Required',
+      'invalidWord': 'invalid word name , only accept alphabets',
+      'invalidPronounce': 'invalid pronounce ,only accept lowercase alphabet and dashes',
+      'maxlength': `Maximum length ${validatorValue.requiredLength}`
+    };
+    return config[validatorName];
+  }
+
+  static wordValidator(control) {
+    if (control.value.match(/^[a-zA-Z]*$/)) {
```
Comment (medhatdawoud) [inline src/app/services/validation.service.ts]: the RegEx should be in an object in the top of the file, not scattered everywhere, put it in an object and use it here and everywhere

File: src/app/services/validation.service.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      return null;
+    } else {
+      return { 'invalidWord': true };
+    }
+  }
+  static pronounceValidator(control) {
+    if (control.value.match(/^[a-z-]*$/)) {
```
Comment (medhatdawoud) [inline src/app/services/validation.service.ts]: same like previous comment

File: src/app/shared/ui-components/error-message/error-message.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (medhatdawoud) [inline src/app/shared/ui-components/error-message/error-message.component.spec.ts]: Is this file autogenerated or you created it ?

File: src/app/shared/ui-components/error-message/error-message.component.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+@Component({
+  selector: 'bw-error-message',
+  template: `<div class="alert alert-danger" *ngIf="errorMessage !== null">{{errorMessage}}</div>`
+})
+export class ErrorMessageComponent {
+  @Input() control: FormControl;
+  constructor() { }
+
+  get errorMessage() {
```
Comment (medhatdawoud) [inline src/app/shared/ui-components/error-message/error-message.component.ts]: why did you use `get` here ?

File: src/app/components/add-word/add-word.component.html (hunk 1 above)
Comment (mahaelbasuony) [reply src/app/components/add-word/add-word.component.html]: fixed

File: src/app/services/validation.service.ts (hunk 2 above)
Comment (mahaelbasuony) [reply src/app/services/validation.service.ts]: fixed

File: src/app/shared/ui-components/error-message/error-message.component.spec.ts (hunk 5 above)
Comment (mahaelbasuony) [reply src/app/shared/ui-components/error-message/error-message.component.spec.ts]: auto-generated

File: src/app/shared/ui-components/error-message/error-message.component.ts (hunk 6 above)
Comment (mahaelbasuony) [reply src/app/shared/ui-components/error-message/error-message.component.ts]: to check if the user touched the form or any action happens to show error messages automatically without creating a listener

File: src/app/services/validation.service.ts (hunk 3 above)
Comment (mahaelbasuony) [reply src/app/services/validation.service.ts]: Fixed

**Pull request merged**